c79126789.lua

```diff
@@ -19,15 +19,17 @@ end
 function c79126789.drtg(e,tp,eg,ep,ev,re,r,rp,chk)
 	if chk==0 then return true end
 	Duel.SetTargetPlayer(tp)
+	Duel.SetTargetParam(1)
 	if tp==rp or tp~=e:GetLabel() then
-		Duel.SetTargetParam(1)
 		Duel.SetOperationInfo(0,CATEGORY_DRAW,nil,0,tp,1)
 	else
-		Duel.SetTargetParam(2)
 		Duel.SetOperationInfo(0,CATEGORY_DRAW,nil,0,tp,2)
 	end
 end
 function c79126789.drop(e,tp,eg,ep,ev,re,r,rp)
 	local p,d=Duel.GetChainInfo(0,CHAININFO_TARGET_PLAYER,CHAININFO_TARGET_PARAM)
-	Duel.Draw(p,d,REASON_EFFECT)
+	if Duel.Draw(p,d,REASON_EFFECT)~=0 and tp==e:GetLabel() then
```

Comment:
the label of this effect isn't reason player

Reply:
https://github.com/Fluorohydride/ygopro-scripts/pull/1740/commits/d8242c7c852af479d64a1b8c8aaac73f4b6f2fbd fixed.